### PR TITLE
Internal: Tests task - Create E2E tests [ED-12050]

### PR DESCRIPTION
**Pull request by hein-obox**
## PR Checklist

- [ ] The commit message follows our guidelines:  https://github.com/elementor/elementor/blob/master/.github/CONTRIBUTING.md

## PR Type
What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] Other... Please describe:

## Summary

This PR can be summarized in the following changelog entry:

*

## Description
An explanation of what is done in this PR

*

## Test instructions
This PR can be tested by following these steps:

*

## Quality assurance

- [ ] I have tested this code to the best of my abilities
- [ ] I have added unittests to verify the code works as intended
- [ ] Docs have been added / updated (for bug fixes / features)

Fixes #

### CODE REVIEW

File: includes/widgets/rating.php [hunk 1]
```diff
@@ -103,7 +103,7 @@ private function add_style_tab() {
 		);
 
 		$this->add_control(
-			'icon_unmasked_color',
+			'icon_unmarked_color',
```
[the remark this answers is not in the document]
Comment (hein-obox) [reply includes/widgets/rating.php:106]: @TzviRabinovitch I found it when I had to write the control name inside the playwright test.

File: tests/playwright/sanity/includes/widgets/rating/style-panel.test.ts [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+
+			await test.step( 'Assert styling', async () => {
+				await expect.soft( await ratingElement.locator( '.e-rating-wrapper' ) ).toHaveCSS( 'gap', '30px' );
+
+				if ( 'active' === iconExperimentState ) {
+					await expect.soft( await ratingElement.locator( '.e-icon >> nth=0' ).locator( 'svg >> nth=1' ) ).toHaveCSS( 'width', '50px' );
+					await expect.soft( await ratingElement.locator( '.e-icon-marked >> nth=0' ).locator( 'svg' ) ).toHaveCSS( 'fill', 'rgb(250, 0, 0)' );
+					await expect.soft( await ratingElement.locator( '.e-icon-unmarked >> nth=0' ).locator( 'svg' ) ).toHaveCSS( 'fill', 'rgb(34, 0, 255)' );
+				} else {
+					// Await page.pause();
```
Comment (TzviRabinovitch) [inline tests/playwright/sanity/includes/widgets/rating/style-panel.test.ts:45]: delete

**Review by TzviRabinovitch — DISMISSED** [review]

File: tests/playwright/sanity/includes/widgets/rating/style-panel.test.ts (hunk 2 above)
Comment (hein-obox) [reply tests/playwright/sanity/includes/widgets/rating/style-panel.test.ts]: @TzviRabinovitch Updated

File: tests/playwright/sanity/includes/widgets/rating/accessibility-structured-data.test.ts [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+			container = await editor.addElement( { elType: 'container' }, 'document' );
+
+		await test.step( 'Add rating widget', async () => {
+			await editor.addWidget( 'rating', container );
+			await editor.setSliderControlValue( 'rating_scale', '7' );
+			await editor.setNumberControlValue( 'rating_value', '3.543' );
+			await editor.publishAndViewPage();
+		} );
+
+		await test.step( '@axe-core/playwright', async () => {
```
Comment (mykytamurzin) [inline tests/playwright/sanity/includes/widgets/rating/accessibility-structured-data.test.ts:30]: wow! automatic accessibility tests!

**Review by murzin-nikita — DISMISSED** [review]

**Comment by mykytamurzin** [comment]
GJ!

File: tests/playwright/sanity/includes/widgets/rating/helper.ts [hunk 4]
```diff
@@ -0,0 +1,27 @@
+import WpAdminPage from '../../../../pages/wp-admin-page';
+
+export async function beforeAll( browser, testInfo, iconExperimentState = 'active' ) {
+	const page = await browser.newPage();
+	const wpAdmin = await new WpAdminPage( page, testInfo );
+
+	await wpAdmin.setExperiments( {
+		container: 'active',
+		rating: 'active',
+		e_font_icon_svg: 'active' === iconExperimentState,
```
Comment (davseve) [inline tests/playwright/sanity/includes/widgets/rating/helper.ts:10]: This comparison is redundant.
```suggestion
		e_font_icon_svg: iconExperimentState,
```

**Review by davseve — CHANGES_REQUESTED** [review]

File: tests/playwright/sanity/includes/widgets/rating/accessibility-structured-data.test.ts [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+			const accessibilityScanResults = await new AxeBuilder( { page } )
+				.include( '.elementor-widget-rating' )
+				.analyze();
+
+			await expect.soft( accessibilityScanResults.violations ).toEqual( [] );
+		} );
```
[the remark this answers is not in the document]
Comment (hein-obox) [reply tests/playwright/sanity/includes/widgets/rating/accessibility-structured-data.test.ts]: Updated.

File: tests/playwright/sanity/includes/widgets/rating/helper.ts (hunk 4 above)
Comment (hein-obox) [reply tests/playwright/sanity/includes/widgets/rating/helper.ts:10]: @davseve For some reason that doesn't work. I am not sure why. It fails both locally and on the CI.

**Review by davseve — APPROVED** [review]

**Pull request merged**